commit fdcb70ac5e974b009e9807bb1e0d63b05a06db34
Author: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date:   Wed Jun 5 02:30:39 2013 +0300

    Declaring default arguments in function implementations is bad C++.
    
    This should fix the build with clang 3.3.

diff --git a/src/FractionPainter.cpp b/src/FractionPainter.cpp
index 48b299a..66362e4 100644
--- src/FractionPainter.cpp
+++ src/FractionPainter.cpp
@@ -28,7 +28,7 @@
 //Added by qt3to4:
 #include <QPaintEvent>
 
-FractionPainter::FractionPainter(QWidget * parent = 0) :
+FractionPainter::FractionPainter(QWidget * parent) :
     FractionBaseWidget(parent)
 {
 #ifdef DEBUG
diff --git a/src/FractionPainter.h b/src/FractionPainter.h
index 9c0c1ab..0fa2a42 100644
--- src/FractionPainter.h
+++ src/FractionPainter.h
@@ -46,7 +46,7 @@ class FractionPainter : public FractionBaseWidget
 
 public:
     /** constructor */
-    FractionPainter(QWidget * parent);
+    FractionPainter(QWidget * parent = 0);
 
     /** destructor */
     ~FractionPainter();
diff --git a/src/fractionbasewidget.cpp b/src/fractionbasewidget.cpp
index ed01d39..8b41bf1 100644
--- src/fractionbasewidget.cpp
+++ src/fractionbasewidget.cpp
@@ -30,7 +30,7 @@
 
 #include "settingsclass.h"
 
-FractionBaseWidget::FractionBaseWidget(QWidget * parent = 0) :
+FractionBaseWidget::FractionBaseWidget(QWidget * parent) :
     QWidget(parent)
 {
 #ifdef DEBUG
diff --git a/src/fractionbasewidget.h b/src/fractionbasewidget.h
index 2973bbb..8d2b078 100644
--- src/fractionbasewidget.h
+++ src/fractionbasewidget.h
@@ -45,7 +45,7 @@ class FractionBaseWidget : public QWidget
 
 public:
     /** constructor */
-    FractionBaseWidget(QWidget * parent);
+    FractionBaseWidget(QWidget * parent = 0);
 
     /** destructor */
     ~FractionBaseWidget();
diff --git a/src/ratiowidget.cpp b/src/ratiowidget.cpp
index 0796411..b2b2e37 100644
--- src/ratiowidget.cpp
+++ src/ratiowidget.cpp
@@ -26,8 +26,8 @@
 //Added by qt3to4:
 #include <QPaintEvent>
 
-RatioWidget::RatioWidget(QWidget * parent = 0,
-                         const ratio para_ratio = ratio()) :
+RatioWidget::RatioWidget(QWidget * parent,
+                         const ratio para_ratio) :
     FractionBaseWidget(parent), m_ratio(para_ratio)
 {
 #ifdef DEBUG
diff --git a/src/ratiowidget.h b/src/ratiowidget.h
index 547ff35..d0d2dd5 100644
--- src/ratiowidget.h
+++ src/ratiowidget.h
@@ -41,7 +41,7 @@ class RatioWidget : public FractionBaseWidget
 
 public:
     /** constructor */
-    RatioWidget(QWidget * parent, const ratio para_ratio);
+    RatioWidget(QWidget * parent = 0, const ratio para_ratio = ratio());
 
     /** destructor */
     ~RatioWidget();
diff --git a/src/taskwidget.cpp b/src/taskwidget.cpp
index 3305189..07efb9a 100644
--- src/taskwidget.cpp
+++ src/taskwidget.cpp
@@ -30,8 +30,8 @@
 //Added by qt3to4:
 #include <QPaintEvent>
 
-TaskWidget::TaskWidget(QWidget * parent = 0,
-                       const task para_task = task()) :
+TaskWidget::TaskWidget(QWidget * parent,
+                       const task para_task) :
     FractionBaseWidget(parent), m_task(para_task)
 {
 #ifdef DEBUG
diff --git a/src/taskwidget.h b/src/taskwidget.h
index 8b8a168..0c3329c 100644
--- src/taskwidget.h
+++ src/taskwidget.h
@@ -51,7 +51,7 @@ class TaskWidget : public FractionBaseWidget
 
 public:
     /** constructor */
-    TaskWidget(QWidget * parent, const task para_task);
+    TaskWidget(QWidget * parent = 0, const task para_task = task());
 
     /** destructor */
     ~TaskWidget();
